File: 0513_med_hot_find-bottom-left-tree-value/solution.java
/**
 * Definition for a binary tree node.
 */

 class TreeNode {
     int val;
     TreeNode left;
     TreeNode right;
     TreeNode() {}
     TreeNode(int val) { this.val = val; }
     TreeNode(int val, TreeNode left, TreeNode right) {
         this.val = val;
         this.left = left;
         this.right = right;
     }
 }
  
  
class Solution {
    int maxDepth = -1;
    int result = 0;
    public int findBottomLeftValue(TreeNode root) {
        if (root == null) {
            return 0;
        }
        findLeftValue(root, 1);
        return result;
    }

    private void findLeftValue(TreeNode node, int depth) {
        if(node == null) {
            return;
        }
        if(node.left == null && node.right == null && depth > maxDepth) {
            maxDepth = depth;
            result = node.val;
        }
        findLeftValue(node.left, depth+1);
        findLeftValue(node.right, depth+1);
    }
}

File: 0513_med_hot_find-bottom-left-tree-value/solution.py
# Definition for a binary tree node.
from collections import deque
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

        
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        if not root:
            return None

        result = root.val
        queue = deque()
        queue.append(root)

        while queue:
            size = len(queue)
            result = queue[0].val
            for _ in range(size):
                node = queue.popleft()
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        
        return result

